lib/solr_pid_dumper.rb

```diff
@@ -0,0 +1,50 @@
+# a thrown-together (but useful and not overly hacky) one-off script for dumping
+# object IDs from solr, to files (json lists), so that other instances can use them for
+# reindexing.  good for things like taking known lists from an existing
+# prod instance and using them for indexing in an upgraded soon-to-be prod instance.
+#
+# to use:
+# * make sure that BASE_DIR exists relative to where this script is executed
+# * make sure OBJ_TYPE_FIELD_NAME is right for the schema of the index you're dumping
+# * make sure OBJ_TYPES is the full list of object types you're interested in
+# * open rails console on the instance you want to dump from
+# * require 'solr_pid_dumper.rb'
+# * call Argo::SolrPidDumper.write_solr_pids_by_obj_type
+# * use your pids for indexing (e.g., as per https://github.com/sul-dlss/argo/blob/develop/INDEXING.md)
+
+module Argo
+  class SolrPidDumper
+    OBJ_TYPE_FIELD_NAME = 'objectType_facet'
+    OBJ_TYPES = %w{workflow agreement adminPolicy collection set}
+    BASE_DIR = 'solr_pid_dumps/'
+
+    def self.solr_pids(q = '*:*')
+      puts "querying solr for indexed pids, (q=#{q})..."
+
+      start = 0
+      solr_pids = []
+      resp = Dor::SearchService.query(q, :sort => 'id asc', :rows => 1000, :start => start, :fl => ['id'])
+      while resp.docs.length > 0
+          solr_pids += resp.docs.map { |doc| doc['id'] }
+          start += 1000
+          resp = Dor::SearchService.query(q, :sort => 'id asc', :rows => 1000, :start => start, :fl => ['id'])
```

Comment:
Line is too long. [110/80]<br>Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.